base/event.jl

```diff
@@ -343,3 +343,50 @@ function Timer(cb::Function, timeout::Real, repeat::Real=0.0)
     yieldto(waiter)
     return t
 end
+
+type TimeoutException <: Exception
+    msg::AbstractString
+
+    TimeoutException() = new("")
+    TimeoutException(msg) = new(msg)
+end
+
+"""
+    @timeout(t, ex, [msg])
+
+Set a timeout of `t` seconds on the execution time of expression `ex`.
+Throws `TimeoutException` with optional error message `msg`
+if the timeout is triggered.
+
+Be aware that the exception raised by the timeout will not abort
+the evaluation of `ex`.
```

Comment:
Ok, looks like this would be done in an appropriate way in a future development.